test/Generator/TypeGeneratorTest.php

```diff
@@ -166,15 +162,48 @@ public function validType()
             ['?object', '?object'],
             ['?Object', '?object'],
             ['?OBJECT', '?object'],
-            ['?mixed', '?\\mixed'],
-            ['?Mixed', '?\\Mixed'],
-            ['?MIXED', '?\\MIXED'],
             ['?resource', '?\\resource'],
             ['?Resource', '?\\Resource'],
             ['?RESOURCE', '?\\RESOURCE'],
             ['?foo_bar', '?\\foo_bar'],
             ["\x80", "\\\x80"],
             ["\x80\\\x80", "\\\x80\\\x80"],
+
+            // Basic union types
+            ['foo|bar', '\\bar|\\foo'],
+            ['\\foo|\\bar', '\\bar|\\foo'],
+            ['foo|string', '\\foo|string'],
+
+            // Capitalization of given types must be preserved
+            ['Foo\\Bar|Baz\\Tab', '\\Baz\\Tab|\\Foo\\Bar'],
+            ['\\Foo\\Bar|\\Baz\\Tab', '\\Baz\\Tab|\\Foo\\Bar'],
+
+            // Union types are sorted
+            ['C|B|D|A', '\\A|\\B|\\C|\\D'],
+            ['string|int|bool|null|float|\\Foo', '\\Foo|bool|int|float|string|null'],
+
+            // Union types may be composed by FQCN and non-FQCN
+            ['\\Foo\\Bar|Baz\\Tab', '\\Baz\\Tab|\\Foo\\Bar'],
+            ['Foo\\Bar|\\Baz\\Tab', '\\Baz\\Tab|\\Foo\\Bar'],
+
+            // Nullable types using `|null` should be equivalent to their `?` counterparts, but
+            // we cannot normalize them until PHP 7 support is dropped.
+            ['foo|null', '\\foo|null'],
+            ['null|foo', '\\foo|null'],
+            ['foo|bar|null', '\\bar|\\foo|null'],
+
+            // The `false` type can only be used in combination with other types
+            ['foo|false', '\\foo|false'],
+            ['string|false', 'string|false'],
+            ['string|false|null', 'string|false|null'],
+
+            // `false` + `null` requires a third type
+            ['Foo|false|null', '\\Foo|false|null'],
+
+            // The `static` type should not be turned into a FQCN
+            ['static', 'static'],
+            ['?static', '?static'],
+            ['static|null', 'static|null'],
```

Comment:
Same comment here. I know that the other cases don't have them, but no reason not to add them to new ones. It also makes them somewhat more self-describing:

```php
'false + null requires third type' => ['Foo|false|null', '\\Foo|false|null'],
```

Reply:
Unsure if I should add it though, because it obscures the default behavior of PHPUnit, which is to render the data provider values on failure :thinking:

Reply:
Decided to skip on this: the types are indeed self-explanatory, and more grouped than singularly documented. Adding the keys is gonna lead to a regression in readability, in this specific case.